### PR TITLE
Add sample Plug-in for demonstrating languages.registerCodeLensProvider API

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

PR adds sample Plug-in that demonstrates the `languages.registerCodeLensProvider` API.
Related PR with the API https://github.com/theia-ide/theia/pull/3414

### CODE REVIEW

File: code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/**
```
Comment (benoitf) [inline code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts]: it seems it is missing copyright headers

Comment (azatsarynnyy) [reply code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts]: added. Thanks

Comment (vitaliy-guliy) [reply code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts]: Should we have copyright headers for all our samples?
I forgot to add it in `tree-view-sample-plugin`

Comment (benoitf) [reply code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts]: I would say yes (we should have a tslint rule for that)

Comment (vitaliy-guliy) [reply code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts]: It looks we haven't. We cannot build our samples from the root.

Comment (benoitf) [reply code-lens-provider-plugin/src/code-lens-provider-plugin-frontend-plugin.ts]: ok maybe let's comment into a new issue

if you want to use lerna, etc to have isolated and also global commands

**Pull request merged**